packages/debugger/src/panels/breakpoints/body.tsx

```diff
@@ -85,61 +109,110 @@ const BreakpointsComponent = ({
  * @param {object} props The component props.
  * @param props.breakpoints The list of breakpoints.
  * @param props.model The model for the breakpoints.
+ * @param props.notebookTracker The notebook tracker.
+ * @param props.config The debugger configuration.
+ * @returns A JSX element.
  */
 const BreakpointCellComponent = ({
   breakpoints,
-  model
+  model,
+  notebookTracker,
+  config
 }: {
   breakpoints: IDebugger.IBreakpoint[];
   model: IDebugger.Model.IBreakpoints;
-}): JSX.Element => {
-  return (
-    <>
-      {breakpoints
-        .sort((a, b) => {
-          return (a.line ?? 0) - (b.line ?? 0);
-        })
-        .map((breakpoint: IDebugger.IBreakpoint, index) => (
-          <BreakpointComponent
-            key={(breakpoint.source?.path ?? '') + index}
-            breakpoint={breakpoint}
-            model={model}
-          />
-        ))}
-    </>
-  );
-};
+  config: IDebugger.IConfig;
+  notebookTracker?: INotebookTracker | undefined;
+}): JSX.Element => (
+  <>
+    {breakpoints
+      .sort((a, b) => {
+        return (a.line ?? 0) - (b.line ?? 0);
+      })
+      .map((breakpoint: IDebugger.IBreakpoint, index) => (
+        <BreakpointComponent
+          key={(breakpoint.source?.path ?? '') + index}
+          breakpoint={breakpoint}
+          model={model}
+          notebookTracker={notebookTracker}
+          config={config}
+        />
+      ))}
+  </>
+);
 
 /**
  * A React Component to display a single breakpoint.
  *
  * @param {object} props The component props.
  * @param props.breakpoint The breakpoint.
  * @param props.model The model for the breakpoints.
+ * @param props.notebookTracker The notebook tracker.
+ * @param props.config The debugger configuration.
+ * @returns A JSX element.
  */
 const BreakpointComponent = ({
   breakpoint,
-  model
+  model,
+  notebookTracker,
+  config
 }: {
   breakpoint: IDebugger.IBreakpoint;
   model: IDebugger.Model.IBreakpoints;
+  config: IDebugger.IConfig;
+  notebookTracker?: INotebookTracker;
 }): JSX.Element => {
-  const moveToEndFirstCharIfSlash = (breakpointSourcePath: string): string => {
-    return breakpointSourcePath[0] === '/'
-      ? breakpointSourcePath.slice(1) + '/'
-      : breakpointSourcePath;
+  const resolveExecutionDisplay = (): string | null => {
+    let display: string | null = null;
+
+    notebookTracker?.forEach(panel => {
+      const kernelName = panel.sessionContext.session?.kernel?.name ?? '';
+      panel.content.widgets.forEach(cell => {
+        if (cell.model.type !== 'code') {
+          return;
+        }
```

Comment:
If we had a way to make those real for-loops, we won't have to loop over all open notebook cells:

```suggestion
    for (const panel of notebookTracker) {
      const kernelName = panel.sessionContext.session?.kernel?.name ?? '';
      for (const cell of panel.content.widgets) {
        if (cell.model.type !== 'code') {
          continue;
        }
        
        // when you found what you're looking for, return!
        // that way you stop the loops as soon as you found it
```

I am not even sure we can do this though?

Reply:
> But maybe IWidgetTracker should implement iterator protocol

Reading the code, I was also thinking about this. Would that be a sensible thing to add as part of this PR? Or do you think it should be done separately?